src/lsp/cobol_lsp/lsp_completion.ml

```diff
@@ -84,25 +92,33 @@ let approx_type_of_usage : usage -> approx_typing_info = function
   | Procedure_pointer
   | Function_pointer _
   | Pointer _
-  | Index
-  | Program_pointer _ -> Numeric
+  | Program_pointer _ -> Pointer
+  | Index -> Numeric
```

Comment:
There's an `INDEX` category that is usually distinct from `NUMERIC` in COBOL, so I think here you should add an `Index` approx type as well.